tls-client/main.cpp

```diff
@@ -256,22 +263,32 @@ class HelloHTTPS {
 
 
         /* Read data out of the socket */
-        ret = mbedtls_ssl_read(&_ssl, (unsigned char *) _buffer, sizeof(_buffer));
+        offset = 0;
+        do {
+            ret = mbedtls_ssl_read(&_ssl, (unsigned char *) _buffer + offset,
+                                   sizeof(_buffer) - offset - 1);
+            if (ret > 0)
+              offset += ret;
+
+            /* Check each of the flags */
+            _buffer[offset] = 0;
+            _got200 = _got200 || strstr(_buffer, HTTPS_OK_STR) != NULL;
+            _gothello = _gothello || strstr(_buffer, HTTPS_HELLO_STR) != NULL;
+        } while ( (!_got200 || !_gothello) &&
+                (ret > 0 || ret == MBEDTLS_ERR_SSL_WANT_READ ||
+                ret == MBEDTLS_ERR_SSL_WANT_WRITE));
```

Comment:
Why are these loops here instead of the `ssl_send()` and `ssl_recv()` functions? It seems that we are having loops over and over. Perhaps it is better to just put them in one place?